Tests/FoundationEssentialsTests/FileManager/FileManagerTests.swift

```diff
@@ -728,6 +728,21 @@ final class FileManagerTests : XCTestCase {
             XCTAssertTrue($0.changeCurrentDirectoryPath(".."))
             XCTAssertEqual(try $0.subpathsOfDirectory(atPath: ".").sorted(), ["bar", "dir", "dir/foo"])
             XCTAssertFalse($0.changeCurrentDirectoryPath("does_not_exist"))
+            
+            XCTAssertThrowsError(try $0.removeItem(atPath: $0.currentDirectoryPath)) {
```

Comment:
Thanks for updating, could you explain a bit about what this test is testing? Since the previous `changeCurrentDirectoryPath` failed I'd expect that `$0.currentDirectoryPath` is currently `..` here - is this removal attempting to test something in specific about `..` here?

Reply:
$0.currentDirectoryPath will only fail after removeItem is executed. In this case, $0.currentDirectoryPath called before removeItem, so there is no issue. Here, deleting $0.currentDirectoryPath removes the current folder, while `..` moves up to the parent directory. Both can actually achieve the same test purpose.

Reply:
> In this case, $0.currentDirectoryPath called before removeItem, so there is no issue. Here, deleting $0.currentDirectoryPath removes the current folder

I think that's the source of my confusion with this particular line - the rest of the test seems ok, but it seems like this line asserts that deleting the parent directory fails, but I wouldn't expect that to be the case on Linux/Darwin where you can remove the current directory. Are you able to explain a little bit more about why this specific `removeItem(atPath:)` fails / what it is testing and how it is different from the `removeItem(atPath:)` which succeeds on non-Windows below?

Reply:
Ohhhhh...Sorry, my mistake. There was an issue when I was merging the code, but the problematic code has already been removed.